raiden-dapp/src/components/HubList.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <div class="hub-list">
+    <span class="hub-list__header">{{ $t('hub-list.header') }}</span>
+    <spinner v-if="loadingHubs" class="hub-list__loading" />
+    <div v-else-if="!loadingHubs && !suggestedHubs" class="hub-list__no-hubs">
+      {{ $t('hub-list.error') }}
+    </div>
+    <div v-for="(suggestedHub, index) in suggestedHubs.slice(0, 3)" v-else :key="index">
+      <div class="hub-list__item">
+        <div class="hub-list__item__icon">
+          <v-img :src="require('@/assets/hub-list/hub-healthy.svg')" />
+        </div>
+        <span class="hub-list__item__address">{{ suggestedHub.address }}</span>
+        <span class="hub-list__item__address-mobile">
+          {{ truncate(suggestedHub.address, 10) }}
+        </span>
+        <v-btn text class="hub-list__item__select-button" @click="selectDeselectHub(index)">
+          <div
+            v-if="hubIsSelected(suggestedHub.address)"
+            class="hub-list__item__select-button__selected-icon"
+          >
+            <v-img :src="require('@/assets/hub-list/hub-selected.svg')" />
+          </div>
+          <span v-else>
+            {{ $t('hub-list.select-button') }}
+          </span>
+        </v-btn>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Emit, Prop } from 'vue-property-decorator';
+import Spinner from '@/components/icons/Spinner.vue';
+import Filters from '@/filters';
+import { SuggestedPartner } from 'raiden-ts/src/services/types';
+
+@Component({
+  components: {
+    Spinner,
+  },
+})
+export default class HubList extends Vue {
+  truncate = Filters.truncate;
+  loadingHubs = true;
+  suggestedHubs: SuggestedPartner[] = [];
+  selectedHub = '';
+
+  @Prop({ required: true })
+  tokenAddress!: string;
+
+  @Emit()
+  selectHub(): string {
+    return this.selectedHub;
+  }
+
+  async mounted() {
+    try {
+      this.suggestedHubs = await this.$raiden.getSuggestedPartners(this.tokenAddress);
+      this.loadingHubs = false;
+    } catch (err) {
+      this.loadingHubs = false;
```

Comment:
You could also but this inside a `finally` block to reduce duplication

Reply:
That's a good idea, I'll put it in a finally block.